### DISCUSSION TITLE
How data is automaticallly updated from 'null' to some actual 'data' ?

### DISCUSSION THREAD

**Discussion by roker15** [vercel/swr#1801] [General]
we call `const { data, error } = useSWR('/api/user/123', fetcher)`
and when not data` !data`, we show loading and when data is fetched we show data. so how does it happen automatically ? Does useSWR use some `setState `internally so that react component automatically updates?

**Comment by sairajchouhan** [comment]
Indeed check this  https://github.com/vercel/swr/blob/7dfd89081d818cba940b7d6bc786e9cdcba24c8e/src/utils/state.ts#L12
